logpuzzle.py

```diff
@@ -21,15 +24,48 @@
 import sys
 import urllib
 import argparse
+import shutil
+
+html_temp = """
+<html>
+<body>
+{}
+</body>
+</html>
+"""
+
+
+def create_html_img(path):
+    """Creates a image tag for html"""
+    return '<img src={}>'.format(path)
+
+
+def sort_funct(url):
+    """"
+    This is a sort custom sorting function and
+    returns the second match group
+    """
+    sort_list = re.search(r'(/p-\w+-)(\w*)', url)
+    return sort_list.group(2)
 
 
 def read_urls(filename):
     """Returns a list of the puzzle urls from the given log file,
     extracting the hostname from the filename itself.
     Screens out duplicate urls and returns the urls sorted into
     increasing order."""
-    # +++your code here+++
-    pass
+    with open(filename) as file:
+        text = file.read()
+        puzzle_list = re.findall(r'GET (\S*puzzle\S*)', text)
+        match_list = re.search(r'(/p-\w+-)(\w*)', text)
+
+        if match_list:
+            match_url = sorted(
+                ['https://code.google.com/' + url for url in set(puzzle_list)])
```

Comment:
Criterion for hostname was to extract it from filename instead of hardcoding the value. Remove hardcoded code.google.com and get this from the domain of the host from the filename.